pkg/provider/azure_instances.go

```diff
@@ -231,6 +231,10 @@ func (az *Cloud) InstanceExists(ctx context.Context, node *v1.Node) (bool, error
 		var err error
 		providerID, err = cloudprovider.GetInstanceProviderID(ctx, az, types.NodeName(node.Name))
 		if err != nil {
+			if errors.Is(err, cloudprovider.InstanceNotFound) {
```

Comment:
Could you also add the same check for L243?

Reply:
https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/pkg/provider/azure_instances.go#L204-L218

The implementation will not return `InstanceNotFound`, the check seems unnecessary?